modules/storages/app/controllers/storages/admin/oauth_clients_controller.rb

```diff
@@ -66,21 +66,27 @@ def create # rubocop:disable Metrics/AbcSize
     call_oauth_clients_create_service
 
     service_result.on_failure do
-      render '/storages/admin/storages/new_oauth_client'
+      respond_to do |format|
+        format.html { render :new }
+      end
     end
 
     service_result.on_success do
       if @storage.provider_type_nextcloud?
         prepare_storage_for_automatic_management_form
 
-        respond_to do |format|
-          format.turbo_stream { render :create }
-        end
-      elsif @storage.provider_type_one_drive?
-        flash[:primer_banner] = { message: I18n.t(:'storages.notice_successful_storage_connection'), scheme: :success }
-        redirect_to admin_settings_storages_path
-      else
-        raise "Unsupported provider type: #{@storage.short_provider_type}"
+        #   respond_to do |format|
+        #     format.turbo_stream { render :create }
+        #   end
+        # elsif @storage.provider_type_one_drive?
+        #   flash[:primer_banner] = { message: I18n.t(:'storages.notice_successful_storage_connection'), scheme: :success }
+        #   redirect_to admin_settings_storages_path
+        # else
+        #   raise "Unsupported provider type: #{@storage.short_provider_type}"
```

Comment:
🟡 this is commented, so we can remove it? you know, were the success flash for onedrive is raised instead?

Reply:
I guess it was moved to the `finish_setup`? And this is called from primer form destination URL?

Reply:
A question related to that, also to @akabiru :

Why does this flash not exist for nextcloud? nc form does not close upon submitting last step.

Reply:
> A question related to that, also to @akabiru :
> 
> Why does this flash not exist for nextcloud? nc form does not close upon submitting last step.

For Nextcloud, the Final Step is automatic management, so the flash is set there: https://github.com/opf/openproject/blob/dev/modules/storages/app/controllers/storages/admin/automatically_managed_project_folders_controller.rb#L77-L81

Reply:
But it does not happen if I click "Complete without it", at least not in my local instance on branch `release/13.1`. Is this intended or a bug?

Reply:
> form destination U

Yes that is right. Technical explanation is difficult. But I can provide if needed.

Reply:
> But it does not happen if I click "Complete without it", at least not in my local instance on branch `release/13.1`. Is this intended or a bug?

The flash is only rendered on first time configuration- it does not render on edit. Let's pair if you're seeing alternate behavior. My local tests confirm this.

Reply:
> 🟡 this is commented, so we can remove it? you know, were the success flash for onedrive is raised instead?

The commented out code should be deleted, we only need to `prepare_storage_for_automatic_management_form` for nextcloud storage

Reply:
we found a proper solution: on viewing the redirect URI in a complete setup (editing a storage), the submit button is no longer enabled, as there is no "setup to complete". this leads to not having a redirect to the list and no flash anymore.